### ISSUE TITLE
sparse_categorical_crossentropy训练时有nan结果

### ISSUE THREAD

**Issue by whalefa1I** [yyccR/yolov5_in_tf2_keras#5]
有的数据会在这行出现nan
https://github.com/yyccR/yolov5_in_tf2_keras/blob/033a1156c1481f4258bf24a4a8215af39682da94/loss.py#L357
查看了input的is_nan，都正常。而且把sparse_categorical_crossentropy换成binary_crossentropy就好了。
请问这两者在这里计算有差别吗，是否可以进行替换

**Comment by yyccR** [comment]
当所给的类别id大于设置的类别数num_classes时就会出现nan, sparse_categorical_crossentropy是稀疏多分类交叉熵，binary_crossentropy是二分类交叉熵，通常是不能替换的，除非你只有1个类别。

**Comment by whalefa1I** [comment]
哦新增一类好像就好了，所以假如我标注的是10类别 ，                
cur_layer_pred_res = tf.keras.layers.Reshape([-1, self.num_class + 5])(cur_layer_pred_res)
这边的num_class = 10 +1 (前景)是吗

**Comment by yyccR** [comment]
不是，看你标注的类别id是从0开始的话，就是num_classes=你的类别数，如果id是从1开始，就是num_classes=你的类别数+1，以此类推
